src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -188,21 +188,22 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
   def createUserPetServiceAccount(user: WorkbenchUser, project: GoogleProject): Future[PetServiceAccount] = {
     val (petSaID, petSaDisplayName) = toPetSAFromUser(user)
 
-    directoryDAO.loadPetServiceAccount(PetServiceAccountId(user.id, project)).flatMap {
-      case Some(pet) => Future.successful(pet)
-      case None =>
-        // First find or create the service account in Google, which generates a unique id and email
-        val petSA = googleIamDAO.getOrCreateServiceAccount(project, petSaID, petSaDisplayName)
-        petSA.flatMap { serviceAccount =>
-          // Set up the service account with the necessary permissions
-          val petServiceAccount = PetServiceAccount(PetServiceAccountId(user.id, project), serviceAccount)
+    // First find or create the service account in Google, which generates a unique id and email
```

Comment:
gpalloc and other "rogue" beings may delete a pet SA in google, leaving a reference to it behind in Sam.

previously we would check to see if the pet SA existed in OpenDJ, and assume everything was cool in google if it existed. since that's not always going to be the case, i've switched up the ordering to first getOrCreate the pet in google and then create the pet in opendj if needed.

per my local logs, this appears to add about ~500ms to each call to get an existing key.

Reply:
Thanks!  Add a comment indicating why we may need to recreate the GooglePet even if it's in OpenDJ, so we don't accidentally "optimize" this in the future